tools/tcpconnect.py

```diff
@@ -107,6 +111,8 @@
     u16 dport = 0;
     bpf_probe_read(&dport, sizeof(dport), &skp->__sk_common.skc_dport);
 
+    FILTER_PORT
```

Comment:
If the bpf_probe_read() really didn't pull a valid port on the entry, then maybe struct sock isn't fully initialized yet. But I think it's ok as it is for now (if it were a little higher frequency, I'd be more interested in making the earlier filter work).